File: ios/Classes/FlutterWebView.swift
import Foundation
import Flutter
import WebKit
 
class FlutterWebView: NSObject, FlutterPlatformView, WKNavigationDelegate {
    private var _nativeWebView: WKWebView
    private var _methodChannel: FlutterMethodChannel
 
    func view() -> UIView {
        return _nativeWebView
    }
 
    init(
        frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?,
        binaryMessenger messenger: FlutterBinaryMessenger
    ) {
        let config = WKWebViewConfiguration()
        _nativeWebView = WKWebView(frame: frame, configuration: config)
        _methodChannel = FlutterMethodChannel(name: "web_view_channel", binaryMessenger: messenger)
 
        super.init()
        _nativeWebView.navigationDelegate = self
        _methodChannel.setMethodCallHandler(onMethodCall)
    }
 
    func onMethodCall(call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "setUrl":
            setUrl(call: call, result: result)
        case "evaluateJavascript":
            evaluateJavascript(call: call, result: result)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
 
    func setUrl(call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let urlString = call.arguments as? String, let url = URL(string: urlString) else {
            result(FlutterError(code: "INVALID_URL", message: "Invalid URL", details: nil))
            return
        }
 
        let request = URLRequest(url: url)
        _nativeWebView.load(request)
    }
 
    func evaluateJavascript(call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let script = call.arguments as? String else {
            result(FlutterError(code: "INVALID_SCRIPT", message: "Invalid JavaScript script", details: nil))
            return
        }
 
        _nativeWebView.evaluateJavaScript(script) { _, error in
            if let error = error {
                result(FlutterError(code: "JAVASCRIPT_ERROR", message: "JavaScript injection failed", details: error.localizedDescription))
            } else {
                result(nil)
            }
        }
    }

    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
        _methodChannel.invokeMethod("onPageFinished", arguments: nil)
    }
}

File: android/src/main/kotlin/com/native/web_view/FlutterWebView.kt
package com.native.web_view

import android.content.Context
import android.view.View
import android.webkit.WebView
import android.webkit.WebViewClient
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.platform.PlatformView


class FlutterWebView internal constructor(
    context: Context,
    messenger: BinaryMessenger,
    id: Int
) :
    PlatformView, MethodCallHandler {
    private val webView: WebView
    private val methodChannel: MethodChannel
    override fun getView(): View {
        return webView
    }

    init {
        webView = WebView(context)
        webView.webViewClient = WebViewClient()
        methodChannel = MethodChannel(messenger, "web_view_$id")
        methodChannel.setMethodCallHandler(this)
    }

    override fun onMethodCall(methodCall: MethodCall, result: MethodChannel.Result) {
        when (methodCall.method) {
            "setUrl" -> setText(methodCall, result)
            else -> result.notImplemented()
        }
    }

    private fun setText(methodCall: MethodCall, result: MethodChannel.Result ) {
        val url = methodCall.arguments as String
        webView.loadUrl(url)
        result.success(null)
    }

    override fun dispose() {
        webView.destroy()
    }

}